### DISCUSSION TITLE
exporting trained model to inference model accuracy crashed

### DISCUSSION THREAD

**Discussion by HawkingRadiation42** [PaddlePaddle/PaddleOCR#12375] [Q&A]
System Environment：windows 11, python: 3.7.9
Version：Paddle：2.7+  PaddleOCR： 

i have read too many issues but couldnt fine any proper solution for it, so asking here hoping for some clue or answers

i have trained my detection paddleocr model using the det_r50_td_tr.yml config file and im getting good results on my dataset 
after exporting to the inference model the accuracy crashes significantly. heard about changing the postprocessing parameters while exporting or prediction but this didnt solve it. 

**another doubt**: in the export_model.py code i couldnt find detection algorithm like db++ in my case has anyone solved this as well?

Command Code：
python tools/export_model.py -c configs/det/det_r50_db++_td_tr.yml -o Global.pretrained_model="./output/det_r50_td_tr_model_completefinetunev2/best_accuracy" Global.save_inference_dir="./output/det_inference_mid_training/"

has anyone here solved this issue?

**Comment by tink2123** [comment]
Your export command is correct and the inference command is ：
**export model**
```
python tools/export_model.py -c configs/det/det_r50_db++_td_tr.yml -o Global.pretrained_model="./output/det_r50_td_tr_model_completefinetunev2/best_accuracy" Global.save_inference_dir="./output/det_inference_mid_training/"
```
**predict:** add --det_box_type args
```
python3 tools/infer/predict_det.py --image_dir="./doc/imgs_en/img623.jpg" --det_model_dir="./output/det_inference_mid_training/" --det_algorithm="DB++" --det_box_type="poly"
```
But you need to modify this line of code https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/tools/infer/predict_det.py#L217
change 
```
dt_boxes = np.array(dt_boxes_new) 
```
to
```
dt_boxes = dt_boxes_new
```

**Comment by HawkingRadiation42** [comment]
thank you for your solution 
but when passing the custom model through the paddleocr class the model accuracy suffers 

custom_ocr = PaddleOCR(
    use_angle_cls=True,
    det_model_dir="D:\hawkingradiation\paddleocr-repo\output\det_inference",
    rec_model_dir="D:\hawkingradiation\paddleocr-repo\inference\en_PP-OCRv4_rec",
    rec_char_dict_path="D:\hawkingradiation\paddleocr-repo\ppocr\utils\en_dict.txt",
    use_gpu=True,
    show_log=False,
  )

1. the paths are correct. 
2. no default model is downloaded during this so the pretrained model is only loaded.

the results are too different in comparison to this way and the predict_det way

any solution for this? or am i making any error here? as the config file isnt passed here so that might be the reason or what?

**Comment by namtranase** [comment]
Hi @HawkingRadiation42 @tink2123, do you find the solution yet?
I face the same issue, the model detects fine with the `python3 tools/infer_det.py` script. But after exporting the model and using `tools/infer/predict_det.py`, it was false to return the text boxes

I'm using:
paddle2onnx            1.0.9
paddleocr              2.7.0.3
paddlepaddle           2.5.2

**Reply by namtranase** [reply]
Found the solution:
https://github.com/PaddlePaddle/PaddleOCR/issues/11644
